hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java

```diff
@@ -791,6 +797,62 @@ public Builder<SystemUndeleteTransactionBody.Builder> systemUndelete() {
         return new Builder<>(TransactionType.SYSTEMUNDELETE, builder);
     }
 
+    public Builder<TokenAirdropTransactionBody.Builder> tokenAirdrop() {
+        var fungibleTokenId = tokenId();
+        var nftTokenId = tokenId();
+
+        // Airdrops that transfer to the account and do not go into the pending airdrop list
+        var tokenTransferList = TokenTransferList.newBuilder()
+                .setToken(fungibleTokenId)
+                .addTransfers(AccountAmount.newBuilder()
+                        .setAccountID(accountId())
+                        .setAmount(-100)
+                        .build())
+                .addTransfers(AccountAmount.newBuilder()
+                        .setAccountID(accountId())
+                        .setAmount(100)
+                        .build());
+        var nftTransferList = TokenTransferList.newBuilder()
+                .setToken(nftTokenId)
+                .addNftTransfers(NftTransfer.newBuilder()
+                        .setSerialNumber(1L)
+                        .setReceiverAccountID(accountId())
+                        .build());
+
+        var fungiblePendingAirdropId = PendingAirdropId.newBuilder()
+                .setSenderId(accountId())
+                .setReceiverId(accountId())
+                .setFungibleTokenType(fungibleTokenId);
+        var fungiblePendingAirdrop = PendingAirdropRecord.newBuilder()
+                .setPendingAirdropId(fungiblePendingAirdropId)
+                .setPendingAirdropValue(
+                        PendingAirdropValue.newBuilder().setAmount(1000L).build());
+        var nftPendingAirdropId = PendingAirdropId.newBuilder()
+                .setSenderId(accountId())
+                .setReceiverId(accountId())
+                .setNonFungibleToken(NftID.newBuilder()
+                        .setTokenID(nftTokenId)
+                        .setSerialNumber(1L)
+                        .build());
+        var nftPendingAirdrop = PendingAirdropRecord.newBuilder().setPendingAirdropId(nftPendingAirdropId);
+
+        return new Builder<>(TransactionType.TOKENAIRDROP, TokenAirdropTransactionBody.newBuilder())
+                .record(r -> r.addTokenTransferLists(tokenTransferList)
+                        .addTokenTransferLists(nftTransferList)
+                        .addNewPendingAirdrops(fungiblePendingAirdrop)
+                        .addNewPendingAirdrops(nftPendingAirdrop));
+    }
+
+    public Builder<TokenCancelAirdropTransactionBody.Builder> tokenCancelAirdrop(PendingAirdropId pendingAirdropId) {
+        var transactionBody = TokenCancelAirdropTransactionBody.newBuilder().addPendingAirdrops(pendingAirdropId);
+        return new Builder<>(TransactionType.TOKENCANCELAIRDROP, transactionBody);
+    }
+
+    public Builder<TokenClaimAirdropTransactionBody.Builder> tokenClaimAirdrop(PendingAirdropId pendingAirdropId) {
```

Comment:
We should have zero-arg methods to create each transaction type. It should take care of generating a typical pending airdrop id. Any customization to add/remove/update those airdrops should be done by the caller. These methods are used by `RecordFileBuilder` via reflection.

```suggestion
    public Builder<TokenCancelAirdropTransactionBody.Builder> tokenCancelAirdrop() {
        var transactionBody = ...
        return new Builder<>(TransactionType.TOKENCANCELAIRDROP, transactionBody);
    }

    public Builder<TokenClaimAirdropTransactionBody.Builder> tokenClaimAirdrop() {
```

Reply:
Thanks I have updated these.